content/developer/writing_plugins/creating-your-own-plugin.md

```diff
@@ -47,15 +46,23 @@ setup(
     version='1.0',
     author='Cloudify',
     packages=['python_webserver'],
-    install_requires=['cloudify-plugins-common>=3.3'],
+    install_requires=['cloudify-common>=4.5.5'],
 )
 {{< /highlight >}}
 
+{{% tip title="Best Practice" %}}
+With the exception of `cloudify-common`, it is strongly recommended that all third-party dependencies
+(specified in the `install_requires` parameter) are version-pinned, rather than using a version range.
+That will guarantee that the plugin is always set up using the very same dependencies, thus avoiding
+cases of plugin code breaking due to incompatible upstream changes.
 
+Alternatively, you could provide version ranges for dependencies, and pin them down during the Wagon
+creation process. Refer to the "Creating Wagons" document for more information.
+{{% /tip %}}
```

Comment:
Our build process looks for a CONSTRAINTS.txt and uses it if it exists. We can tell the user to pin them there if you want.

Reply:
@EarthmanT this document is public-facing. So, mentioning `constraints.txt` is only relevant if our build process can (and is intended to) be used by others. Is that the case? (I really don't know)